cleanlab/filter.py

```diff
@@ -398,6 +403,194 @@ class 0, 1, ..., K-1.
+      If ``True``, prints when multiprocessing happens.
+
+    Returns
+    -------
+    label_issues : np.ndarray
+      If `return_indices_ranked_by` left unspecified, returns a boolean **mask** for the entire dataset
+      where ``True`` represents a label issue and ``False`` represents an example that is
+      accurately labeled with high confidence.
+      If `return_indices_ranked_by` is specified, returns a shorter array of **indices** of examples identified to have
+      label issues (i.e. those indices where the mask would be ``True``), sorting by likelihood that the corresponding label is correct is not supported yet.
+
+      Note
+      ----
+      Obtain the *indices* of label issues in your dataset by setting
+      `return_indices_ranked_by`.
+
+    """
+    num_classes = get_num_classes(labels=labels, pred_probs=pred_probs)
+    y_one = int2onehot(labels)
+    if return_indices_ranked_by is None:
+        bissues = np.zeros(y_one.shape).astype(bool)
+    else:
+        label_issues_list = []
+    labels_list = []
+    pred_probs_list = []
+    if confident_joint is not None:
+        confident_joint_shape = confident_joint.shape
+        if confident_joint_shape == (num_classes, num_classes):
+            warnings.warn(
+                f"The new recommended format for confident_joint in multi_label settings is (num_classes,2,2) (as output by compute_confident_joint(...,multi_label=True)). Your k x k confident_joint in the old format is being ignored."
+            )
+            confident_joint = None
+        elif confident_joint_shape != (num_classes, 2, 2):
+            raise ValueError("confident_joint should be of shape (num_classes,2, 2)")
+    for class_num in range(0, num_classes):
+        pred_probabilitites = _binarize_pred_probs_slice(pred_probs, class_num)
+        if confident_joint is None:
+            conf = None
+        else:
+            conf = confident_joint[class_num]
+        binary_label_issues = find_label_issues(
+            labels=y_one[:, class_num],
+            pred_probs=pred_probabilitites,
+            return_indices_ranked_by=return_indices_ranked_by,
+            frac_noise=frac_noise,
+            rank_by_kwargs=rank_by_kwargs,
+            filter_by=filter_by,
+            multi_label=False,
+            num_to_remove_per_class=num_to_remove_per_class,
+            min_examples_per_class=min_examples_per_class,
+            confident_joint=conf,
+            n_jobs=n_jobs,
+            verbose=verbose,
+        )
+
+        if return_indices_ranked_by is None:
+            bissues[:, class_num] = binary_label_issues
+        else:
+            label_issues_list.append(binary_label_issues)
+            labels_list.append(y_one[:, class_num])
+            pred_probs_list.append(pred_probabilitites)
+    if return_indices_ranked_by is None:
+        return bissues.sum(axis=1) >= 1
+    else:
+        label_issues_idx = reduce(np.union1d, label_issues_list).astype(np.int32)
```

Comment:
When I run your test code from your PR description, but I add in: `return_indices_ranked_by='self_confidence', then I get back: `array([4], dtype=int32)`, presumably because of this line.

However that doesn't match output of `find_label_issues(..., multi_label=False, return_indices_ranked_by='self_confidence')` which would typically just look like this: `array([4])`

Can we just make the multi-label case match the existing return type from the regular non-multilabel case? I don't think returning two different types is a good idea